yarn-project/aztec-node/src/aztec-node/server.ts

```
@@ -383,24 +379,31 @@ export class AztecNodeService implements AztecNode, AztecNodeAdmin, Traceable {
         blockBuilder,
         config,
       );
-      await epochPruneWatcher.start();
       watchers.push(epochPruneWatcher);
     }
 
     // We assume we want to slash for invalid attestations unless all max penalties are set to 0
     let attestationsBlockWatcher: AttestationsBlockWatcher | undefined;
     if (config.slashProposeInvalidAttestationsPenalty > 0n || config.slashAttestDescendantOfInvalidPenalty > 0n) {
       attestationsBlockWatcher = new AttestationsBlockWatcher(archiver, epochCache, config);
-      await attestationsBlockWatcher.start();
       watchers.push(attestationsBlockWatcher);
     }
 
-    log.verbose(`All Aztec Node subsystems synced`);
+    // Start p2p-related services once the archiver has completed sync
+    void archiver
+      .waitForInitialSync()
+      .then(async () => {
+        await p2pClient.start();
+        await validatorsSentinel?.start();
+        await epochPruneWatcher?.start();
+        await attestationsBlockWatcher?.start();
+        log.info(`All p2p services started`);
+      })
+      .catch(err => log.error('Failed to start p2p services after archiver sync', err));
```

Could these throw? Right now we're catching the error and swallowing it here. The node will continue running regardless of `skipArchiverInitialSync`

---

They shouldn't throw, that's there just to prevent an unhandled exception in the odd case they do throw

---

should it `process.exit(1)` if they do throw? Anyway, approved. We can tackle this in a separte PR if needed

---

Created this issue so we don't forget about it [A-205: chat about potentially stopping the node if its components fail to start](https://linear.app/aztec-labs/issue/A-205/chat-about-potentially-stopping-the-node-if-its-components-fail-to)